typing/typecore.ml

```diff
@@ -5056,9 +5056,10 @@ let report_error env ppf = function
         (function ppf ->
            fprintf ppf "but is here used with type");
       if b then
-        fprintf ppf ".@.@[<hov>%s@ %s@]"
+        fprintf ppf ".@.@[<hov>%s@ %s@ %s@]"
           "This simple coercion was not fully general."
-          "Consider using a double coercion."
+          "Hint: Consider using a double coercion"
+          "(of the form: `(foo : ty1 :> ty2)')."
```

Comment:
I think removing the parentheses around `of the form: '...'` would make the whole thing more readable.

Reply:
like that?